Lazydocs: API Reference doc refresh (#1424)
wandb/docs: content/en/launch/integration-guides/dagster.md

W&B supports multiple Pickle-based serialization modules ([pickle](https://docs.python.org/3/library/pickle.html), [dill](https://github.com/uqfoundation/dill), [cloudpickle](https://github.com/cloudpipe/cloudpickle), [joblib](https://github.com/joblib/joblib)). You can also use more advanced serialization like [ONNX](https://onnx.ai/) or [PMML](https://en.wikipedia.org/wiki/Predictive_Model_Markup_Language). Please refer to the [Serialization]({{< relref "#serialization-configuration" >}}) section for more information.
{{% /tab %}}
{{% tab "W&B Object" %}}

> Any native W&B object (e.g [Table]({{< relref "/ref/python/sdk/data-types/table.md" >}}), [Image]({{< relref "/ref/python/sdk/data-types/image.md" >}})) is added to an Artifact created by the integration. Here’s an example using a Table.

Reviewer: ```suggestion
Any W&B object, such as a [Table]({{< relref "/ref/python/sdk/data-types/table.md" >}}) or [Image]({{< relref "/ref/python/sdk/data-types/image.md" >}}), is added to an Artifact created by the integration. This example adds a Table to an Artifact:
```

* `type`: (str) The type of the artifact, which is used to organize and differentiate artifacts. Common types include dataset or model, but you can use any string containing letters, numbers, underscores, hyphens, and dots. Required when the output is not already an Artifact.
* `description`: (str) Free text that offers a description of the artifact. The description is markdown rendered in the UI, so this is a good place to place tables, links, etc.
* `aliases`: (list[str]) An array containing one or more aliases you want to apply on the Artifact. The integration will also add the “latest” tag to that list whether it’s set or not. This is an effective way for you to manage versioning of models and datasets.

> * [`add_dirs`]({{< relref "/ref/python/sdk/classes/artifact#add_dir" >}}): (list[dict[str, Any]]): An array containing configuration for each local directory to include in the Artifact. It supports the same arguments as the homonymous method in the SDK.

Reviewer: ```suggestion
* [`add_dirs`]({{< relref "/ref/python/sdk/classes/artifact#add_dir" >}}): (list[dict[str, Any]]): An array containing configuration for each local directory to include in the Artifact. It supports the same arguments as the same named method in the SDK.
```
"homonymous" is a crazy high reading comprehension level word. More instances below.

Author: Oh gosh. This must've slipped the cracks when the eng created the PR way back when.

## Best practices

1. Use the IO Manager to read and write Artifacts. 

> You should never need to use [`Artifact.download()`]({{< relref "/ref/python/sdk/classes/artifact#download" >}}) or [`Run.log_artifact()`]({{< relref "/ref/python/sdk/classes/run#log_artifact" >}}) directly. Those methods are handled by integration. Simply return the data you wish to store in Artifact and let the integration do the rest. This will provide better lineage for the Artifact in W&B.

Reviewer: ```suggestion
Avoid using [`Artifact.download()`]({{< relref "/ref/python/sdk/classes/artifact#download" >}}) or [`Run.log_artifact()`]({{< relref "/ref/python/sdk/classes/run#log_artifact" >}}) directly. Those methods are handled by integration. Instead, return the data you want to store in the Artifact and let the integration do the rest. This approach provides better lineage for the Artifact.
```
